modules/templates/templates/templates.tsx

```diff
@@ -0,0 +1,250 @@
+          />
+        ) : (
+          <div
+            onClick={() =>
+              router.push(`/app/template-detail?id=${template.id}`)
+            }
+            className="flex items-center"
+            style={{ gap: "1vh", cursor: "pointer" }}
+          >
+            <span>{template.name}</span>
+            <FaRegFolderOpen size={17.5} />
+          </div>
+        ),
+    },
+    {
+      title: t("templates.date"),
+      dataIndex: "createdAt",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="date"
+            value={
+              new Date(editingTemplate.createdAt).toISOString().split("T")[0]
+            }
+            onChange={(e) => handleInputChange("createdAt", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <span>{new Date(template.createdAt).toLocaleDateString()}</span>
+        ),
+    },
+    {
+      title: t("templates.preview"),
+      dataIndex: "preview",
+      render: (template: GenesisTemplate) =>
+        editingTemplate && editingTemplate.id === template.id ? (
+          <input
+            type="text"
+            value={editingTemplate.preview}
+            onChange={(e) => handleInputChange("preview", e.target.value)}
+            className={styles.input}
+          />
+        ) : (
+          <span>{template.preview}</span>
+        ),
+    },
+    {
+      title: t("templates.action"),
+      dataIndex: "id",
+      render: (template: GenesisTemplate) => (
+        <>
+          {editingTemplate && editingTemplate.id === template.id ? (
+            <button onClick={handleSave} className={styles.actionButton}>
+              <FaSave style={{ color: "#59DBBC" }} />
+            </button>
+          ) : (
+            <>
+              <button
+                onClick={() => handleEdit(template)}
+                className={styles.actionButton}
+                style={{ marginRight: "4vh" }}
+              >
+                <FaEdit style={{ color: "#59DBBC" }} />
+              </button>
+              <button
+                onClick={() => handleDelete(template.id)}
+                className={styles.actionButton}
+              >
+                <FaTrash style={{ color: "#e53e3e" }} />
+              </button>
+            </>
+          )}
+        </>
+      ),
+    },
+  ];
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <h1 className={styles.title}>{t("templates.all_templates")}</h1>
+        <button className={styles.addButton} onClick={handleAddTemplate}>
+          <FaPlus /> {t("templates.create")}
+        </button>
+      </div>
+      <Table data={templates} columns={columns} isLoading={isLoading} />
+      <Pagination
+        currentPage={currentPage}
+        totalPages={totalPages}
+        totalRecords={totalRecords}
+        onPageChange={handlePageChange}
+      />
```

Comment:
I though pagination is insite the table.  
It only needs to properties  `currentPage` and `totalRecords` (total Pages can be deduced of the page size which can be a optional property)

Reply:
No

Reply:
?

Disagreeing is ok, but need to justify